### DISCUSSION TITLE
Tycho 2.7.0 takes hours resolving missing classpath entries

### DISCUSSION THREAD

**Discussion by wimjongman** [eclipse-tycho/tycho#764] [General]
I have upgraded to 2.7.0. I use Jenkins and Tycho 2.7.0 and Maven 3.6.3

It turns out I have this plug-in that lists additional local jars (fat jar pattern) on the classpath but the jars are not part of the build.properties.

For every module that references this plug-in, Tycho 2.7.0 spins for 5 minutes. The build time is increased from 15 minutes to 90 minutes.

I can fix it in my build but I thought I mention it anyway:

`
[WARNING] Bundle-ClassPath entry lib/httpcomponents-client-4.5.7/lib/jna-4.5.2.jar does not exist in /opt/jenkins-slave/workspace/MiWorkplace_Update_Site/.repository/p2/osgi/bundle/com.remainsoftware.common.licensing/1.0.0.202203140903/com.remainsoftware.common.licensing-1.0.0.202203140903.jar
`

After this message, Tycho 2.7.0 chokes for several minutes

**Comment by laeubi** [comment]
Can you reproduce this with a local build and collect some performance statistics?
By the way the message is from a part what causes problem in parallel build : https://github.com/eclipse/tycho/issues/663 and I see there is a locking attempt with a timeout of 5 minutes.... does your build executes in parallel or concurrently with other jobs sharing the same m2 local repository.
